onchain/rollups/test/foundry/consensus/quorum/QuorumFactory.t.sol

```diff
@@ -0,0 +1,121 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+/// @title Quorum Factory Test
+pragma solidity ^0.8.8;
+
+import {Test} from "forge-std/Test.sol";
+import {QuorumFactory} from "contracts/consensus/quorum/QuorumFactory.sol";
+import {Quorum} from "contracts/consensus/quorum/Quorum.sol";
+import {IHistory} from "contracts/history/IHistory.sol";
+import {Vm} from "forge-std/Vm.sol";
+
+contract QuorumFactoryTest is Test {
+    QuorumFactory factory;
+    address[] validators;
+    uint256[] shares;
+    uint256 numValidator;
```

Comment:
This state variable can be removed, and any usage of it can be substituted by the length of either array.

Reply:
Yes we can. Although it's more concise to say `numValidator` compared with `_validators.length`. So even after removing this state variable, I would probably still create a local variable whichever test it appears in. And since it's a test contract, we don't need to worry about gas optimization. So one state variable is easier than potentially multiple local variables? What do you think?

Reply:
I think we should only store the array in storage and access its length locally in each function, this way we avoid the two variables going out-of-sync.